fdbclient/GlobalConfig.actor.cpp

```diff
@@ -128,45 +155,50 @@ void GlobalConfig::erase(KeyRangeRef range) {
-
-	state Key migratedKey("\xff\x02/fdbClientInfo/migrated/"_sr);
-	state Optional<Value> migrated = wait(tr->get(migratedKey));
-	if (migrated.present()) {
-		// Already performed migration.
-		return Void();
-	}
+	state Reference<ReadYourWritesTransaction> tr;
+	loop {
+		try {
+			tr = makeReference<ReadYourWritesTransaction>(Database(Reference<DatabaseContext>::addRef(self->cx)));
+			wait(delay(0));
+			tr->setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
+
+			state Key migratedKey("\xff\x02/fdbClientInfo/migrated/"_sr);
+			state Optional<Value> migrated = wait(tr->get(migratedKey));
+			if (migrated.present()) {
+				// Already performed migration.
+				return Void();
+			}
 
-	state Optional<Value> sampleRate = wait(tr->get(Key("\xff\x02/fdbClientInfo/client_txn_sample_rate/"_sr)));
-	state Optional<Value> sizeLimit = wait(tr->get(Key("\xff\x02/fdbClientInfo/client_txn_size_limit/"_sr)));
+			state Optional<Value> sampleRate = wait(tr->get(Key("\xff\x02/fdbClientInfo/client_txn_sample_rate/"_sr)));
+			state Optional<Value> sizeLimit = wait(tr->get(Key("\xff\x02/fdbClientInfo/client_txn_size_limit/"_sr)));
+
+			tr->setOption(FDBTransactionOptions::SPECIAL_KEY_SPACE_ENABLE_WRITES);
+			// The value doesn't matter too much, as long as the key is set.
+			tr->set(migratedKey.contents(), "1"_sr);
+			if (sampleRate.present()) {
+				const double sampleRateDbl =
+				    BinaryReader::fromStringRef<double>(sampleRate.get().contents(), Unversioned());
+				Tuple rate = Tuple().appendDouble(sampleRateDbl);
+				tr->set(GlobalConfig::prefixedKey(fdbClientInfoTxnSampleRate), rate.pack());
+			}
+			if (sizeLimit.present()) {
+				const int64_t sizeLimitInt =
+				    BinaryReader::fromStringRef<int64_t>(sizeLimit.get().contents(), Unversioned());
+				Tuple size = Tuple().append(sizeLimitInt);
+				tr->set(GlobalConfig::prefixedKey(fdbClientInfoTxnSizeLimit), size.pack());
+			}
 
-	try {
-		tr->setOption(FDBTransactionOptions::SPECIAL_KEY_SPACE_ENABLE_WRITES);
-		// The value doesn't matter too much, as long as the key is set.
-		tr->set(migratedKey.contents(), "1"_sr);
-		if (sampleRate.present()) {
-			const double sampleRateDbl =
-			    BinaryReader::fromStringRef<double>(sampleRate.get().contents(), Unversioned());
-			Tuple rate = Tuple().appendDouble(sampleRateDbl);
-			tr->set(GlobalConfig::prefixedKey(fdbClientInfoTxnSampleRate), rate.pack());
-		}
-		if (sizeLimit.present()) {
-			const int64_t sizeLimitInt =
-			    BinaryReader::fromStringRef<int64_t>(sizeLimit.get().contents(), Unversioned());
-			Tuple size = Tuple().append(sizeLimitInt);
-			tr->set(GlobalConfig::prefixedKey(fdbClientInfoTxnSizeLimit), size.pack());
+			wait(tr->commit());
+			break;
+		} catch (Error& e) {
+			// If multiple fdbserver processes are started at once, they will all
+			// attempt this migration at the same time, sometimes resulting in
+			// aborts due to conflicts. Purposefully avoid retrying, making this
+			// migration best-effort.
```

Comment:
This comment is no longer true.

I think earlier discussion is that we should add another `loop` outside this one, to handle non-retryable errors.

Reply:
Retrying a non-retryable error doesn't seem like the best way to handle it. What about catching the non-retryable error but instead of re-throwing it, returning as if the function succeeded (this avoids blocking). This could mean that in some cases the migration fails, but only if the migration gets a non-retryable error.